app/main.py

```diff
@@ -1,3 +1,50 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+
+        return Distance(km=self.km + other)
+
+    def __iadd__(self, other: Distance | int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+
+        self.km += other
+        return self
+
+    def __mul__(self, other: int) -> Distance:
+        return Distance(
+            km=self.km * other
+        )
+
+    def __lt__(self, other: Distance) -> bool:
+        return self.km < other
+
+    def __gt__(self, other: Distance) -> bool:
+        return self.km > other
+
+    def __eq__(self, other: Distance) -> bool:
+        return self.km == other
+
+    def __le__(self, other: Distance) -> bool:
+        return self.km <= other
+
+    def __ge__(self, other: Distance) -> bool:
+        return self.km >= other
```

Comment:
These methods are wrong.
1. `other` can be as Distance as int
2. You need to change methods' implementation to support either int and Distance